### PR TITLE
Add ProxyConfig subcommand

### LINKED ISSUE #47: Command to list, show, and promote proxy-configs
Get a list of proxy configs

Display info about the proxy config

Promote an existing proxy config to the production environment

### PR DESCRIPTION
**Pull request by miguelsorianod**
Fixes #47

### CODE REVIEW

File: lib/3scale_toolbox/commands/proxy_config_command/list_command.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          def service_ref
+            arguments[:service_ref]
+          end
+
+          def proxyconfigs
+            @proxyconfigs ||= list_proxy_configs
+          end
+
+          def list_proxy_configs
+            proxyconfigs = remote.proxy_config_list(service.id, proxy_config_environment)
```
Comment (eguzki) [inline lib/3scale_toolbox/commands/proxy_config_command/list_command.rb]: Consider having in `Entities::Service.proxy_configs` method and from here call

`service.proxy_configs(proxy_config_environment)`

When requesting a list of children of a service, we can do it using Service entity and return instances of `Entitities::ProxyConfig`

If you do not like this change, at least you should check for errros. All calls to `remote` object should be checked. 

That makes me think, we should wrap remote object and check for errors, so commands or entities do not have to do it. wdyt?

File: lib/3scale_toolbox/entities/proxy_config.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+module ThreeScaleToolbox
+  module Entities
+    class ProxyConfig
+      class << self
+        def find(service:, environment:, version:)
+            new(service: service, environment: environment, version: version).tap(&:attrs)
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply lib/3scale_toolbox/entities/proxy_config.rb]: fixed

File: lib/3scale_toolbox/commands/proxy_config_command/promote_command.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            puts "Proxy Configuration promoted to '#{to_env}'"
+          end
+
+          private
+
+          def remote
+            @remote ||= threescale_client(arguments[:remote])
+          end
+
+          def proxy_config
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply lib/3scale_toolbox/commands/proxy_config_command/promote_command.rb]: changed

File: lib/3scale_toolbox/commands/proxy_config_command/list_command.rb (hunk 1 above)
Comment (miguelsorianod) [reply lib/3scale_toolbox/commands/proxy_config_command/list_command.rb]: I implemented the proxy_configs method.
We can open an issue for the remote redesign.

File: lib/3scale_toolbox/commands/proxy_config_command/promote_command.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            @proxy_config ||= find_proxy_config_latest
+          end
+
+          def find_proxy_config_latest
+            proxy_config = Entities::ProxyConfig.find_latest(service: service, environment: from_env).tap do |pc|
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply lib/3scale_toolbox/commands/proxy_config_command/promote_command.rb]: fixed

**Comment by miguelsorianod** [comment]
Added documentation

**Pull request merged**